File: app/src/main/java/com/ternaryop/photoshelf/service/BootServiceReceiver.java
package com.ternaryop.photoshelf.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

// http://stackoverflow.com/questions/7344897/autostart-android-service
// adb shell am broadcast -a android.intent.action.BOOT_COMPLETED

public class BootServiceReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(final Context context, final Intent intent) {
        if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            // don't slow the boot running immediately the service, wait 5 seconds
            AlarmBroadcastReceiver.createBirthdayAlarm(context, System.currentTimeMillis() + 5000);
            AlarmBroadcastReceiver.createExportAlarm(context, System.currentTimeMillis() + 15000);
        }
    }
}

File: app/src/main/java/com/ternaryop/photoshelf/service/BootServiceReceiver.kt
package com.ternaryop.photoshelf.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

// https://stackoverflow.com/questions/11325920/how-to-test-boot-completed-broadcast-receiver-in-emulator#comment57079528_30407531
// adb shell am broadcast -a android.intent.action.BOOT_COMPLETED -p com.ternaryop.photoshelf.debug

class BootServiceReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            PhotoShelfJobService.scheduleBirthday(context)
            PhotoShelfJobService.scheduleExport(context)
        }
    }
}
